### PR TITLE
Replace local H2 db by AWS DynamoDB

### PR DESCRIPTION
**Pull request by menabebawy**
- replaced h2 db by aws dynamodb
- unit tests ✅

### CODE REVIEW

File: api/src/main/java/com/mybooks/api/repository/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.mybooks.api.repository;
+
+import com.mybooks.api.model.Book;
+import org.socialsignin.spring.data.dynamodb.repository.EnableScan;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+
+@EnableScan
```
Comment (malirizk) [inline api/src/main/java/com/mybooks/api/repository/BookRepository.java]: why this is used?

File: api/src/main/java/com/mybooks/api/service/BookServiceImpl.java [hunk 2]
```diff
@@ -2,20 +2,22 @@
...
-import lombok.AllArgsConstructor;
+import com.mybooks.api.repository.BookRepository;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
-@AllArgsConstructor
 @Service
-public class BookServiceImp implements BookService {
+public class BookServiceImpl implements BookService {
```
Comment (malirizk) [inline api/src/main/java/com/mybooks/api/service/BookServiceImpl.java:11]: Aywa kda enta 2albt Java

File: api/src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# AWS properties
```
Comment (malirizk) [inline api/src/main/resources/application.yml:1]: Try to read about Spring Profiling

File: api/src/test/java/com/mybooks/api/controller/BookControllerTest.java [hunk 4]
```diff
@@ -91,21 +90,21 @@ public void updateBook_success() throws Exception {
         Book updatedBook = getMockBook();
         updatedBook.setTitle("New Title");
 
-        when(bookService.updateBook(any(Book.class), any(String.class))).thenReturn(updatedBook);
+        Mockito.when(bookService.updateBook(ArgumentMatchers.any(Book.class), ArgumentMatchers.any(String.class))).thenReturn(updatedBook);
```
Comment (malirizk) [inline api/src/test/java/com/mybooks/api/controller/BookControllerTest.java]: why  you added `Mockito.`? you can use static import instead!!

**Review by malirizk — APPROVED** [review]

File: api/src/main/java/com/mybooks/api/repository/BookRepository.java (hunk 1 above)
Comment (menabebawy) [reply api/src/main/java/com/mybooks/api/repository/BookRepository.java]: removed

**Pull request merged**